app/grandchallenge/algorithms/views.py

```diff
@@ -1091,3 +1095,61 @@ def get(self, *_, **__):
                 filename=f"{algorithm.slug}-template.zip",
                 content_type="application/zip",
             )
+
+
+class AlgorithmInterfacePermissionMixin(
+    VerificationRequiredMixin, AccessMixin
```

Comment:
This is stacking `UserPassesTestMixin` on top of `AccessMixin` which might lead to trouble. Why is verification required here?

Reply:
We check for verification on all other algorithm views as well, hence the inclusion here. We do already have a similar stacking of `UserPassesTestMixin` on top of `AccessMixin` in other places (e.g. `MethodCreate` where we use the `VerificationRequredMixin` and `ActiveChallengeRequiredMixin`). My tests suggest this works and is not a problem, but I could also include the verificaiton check in the dispatch method directly to be sure, or we drop the verification requirement, but that would not be in line with the rest of the algorithm create and update logic.

Reply:
Verification should only be required for container uploads, which is why it is in `MethodCreate`. Changing the algorithm settings like this should not require verification. Verification is only there to protect our resources somewhat. Looking through the history:

- AlgorithmImageCreate: Should stay to prevent spam containers (consumes storage)
- JobCreate: Should stay to prevent spam jobs (consumes storage and compute)
- AlgorithmRepositoryUpdate: Should stay to prevent spam containers (consumes storage)
- AlgorithmModelCreate: Should stay to prevent spam models (consumes storage)

However:

- AlgorithmCreate (#1979): This was added when we allowed all challenge users to create algorithms, we now have other limits in place via the phase algorithm create so I do not think that this is necessary anymore.
- AlgorithmUpdate (#2030): Added for GitHub checking, those settings are no longer there and are now in AlgorithmRepositoryUpdate. I don't think there is anything in there that now requires verification.
- AlgorithmDescriptionUpdate (#2171): Copied from #2030 when the settings were split, again I don't think this is necessary.
- Also in AlgorithmInterfacePermissionMixin: I don't see a reason to limit it to verified users.

Reply:
Ok, I just wanted to stay within the existing pattern/logic, but your argumentation is convincing. I will remove the verification check.

Reply:
Added #3744 for cleanup of the existing views.